### PR TITLE
Fix bug that produces duplicate vertices

### PR DESCRIPTION
**Pull request by pmneila**

### CODE REVIEW

File: mcubes/src/marchingcubes.h [hunk 1]
```diff
@@ -74,111 +89,117 @@ void marching_cubes(const vector3& lower, const vector3& upper,
                 if(edges & 0x040)
```
Comment (asyniakov) [inline mcubes/src/marchingcubes.h]: vector is very costly here. it's better to rewrite like this one: ```
      std::int32_t indices[12] = {
                        -1, -1, -1, -1,
                        -1, -1, -1, -1,
                        -1, -1, -1, -1,
                    };
```

Comment (pmneila) [reply mcubes/src/marchingcubes.h]: You are absolutely right. I will fix this.

Comment (pmneila) [reply mcubes/src/marchingcubes.h]: It's fixed now. Thank you!

File: mcubes/src/marchingcubes.h [hunk 2]
```diff
@@ -74,111 +89,117 @@ void marching_cubes(const vector3& lower, const vector3& upper,
...
                 {
                     indices[6] = vertices.size() / 3;
-                    shared_indices[i_mod_2*yz3 + j*z3 + k*3 + 0] = indices[6];
+                    shared_indices_x[i_mod_2_inv*numyz + (j+1)*numz + (k+1)] = indices[6];
                     mc_add_vertex(x_dx, y_dy, z_dz, x, 0, v[6], v[7], isovalue, &vertices);
```
Comment (asyniakov) [inline mcubes/src/marchingcubes.h]: also I believe you can do reserve for vertices to speed up some things.

Comment (asyniakov) [reply mcubes/src/marchingcubes.h]: and for polygons as well

**Pull request merged**